### PR TITLE
gh-114728: Fix documentation for comparison of objects in datetime module

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-114728

----
📚 Documentation preview 📚: https://cpython-previews--114749.org.readthedocs.build/en/114749/library/datetime.html

### CODE REVIEW

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -606,8 +583,13 @@ Supported operations:
...
 +-------------------------------+----------------------------------------------+
-| ``date1 < date2``             | *date1* is considered less than *date2* when |
-|                               | *date1* precedes *date2* in time. (4)        |
+| ``date1 == date2``            | Equality comparison. (4)                     |
+| ``date1 != date2``            |                                              |
++-------------------------------+----------------------------------------------+
+| ``date1 < date2``             | Order comparison. (5)                        |
+| ``date1 > date2``             |                                              |
+| ``date1 <= date2``            |                                              |
+| ``date1 >= date2``            |                                              |
```
Comment (hugovk) [inline Doc/library/datetime.rst]: This ^ renders like:

image

This:


```suggestion
| ``date1 == date2``            | Equality comparison. (4)                     |
|                               |                                              |
| ``date1 != date2``            |                                              |
+-------------------------------+----------------------------------------------+
| ``date1 < date2``             | Order comparison. (5)                        |
|                               |                                              |
| ``date1 > date2``             |                                              |
|                               |                                              |
| ``date1 <= date2``            |                                              |
|                               |                                              |
| ``date1 >= date2``            |                                              |
```

Renders like:

image

And this:

```suggestion
| ``date1 == date2``            | Equality comparison. (4)                     |
+-------------------------------+                                              +
| ``date1 != date2``            |                                              |
+-------------------------------+----------------------------------------------+
| ``date1 < date2``             | Order comparison. (5)                        |
+-------------------------------+                                              +
| ``date1 > date2``             |                                              |
+-------------------------------+                                              +
| ``date1 <= date2``            |                                              |
+-------------------------------+                                              +
| ``date1 >= date2``            |                                              |
```

Renders like:

image

Same applies to the `datetime` table below.

Comment (serhiy-storchaka) [reply Doc/library/datetime.rst]: Good catch!

I fixed this using other method.

**Review by hugovk — APPROVED** [review]
Looks good, would be nice if Paul or Alexander get a chance to review too.

File: Doc/library/datetime.rst [hunk 2]
```diff
@@ -627,15 +609,20 @@ Notes:
    timedelta.microseconds are 0, and date2 + timedelta == date1 after.
 
 (4)
+   :class:`date` objects are equal if they represent the same date.
+
+   :class:`!date` objects that are not also :class:`.datetime` instances
+   are never equal to :class:`!datetime` objects, even if they represent
+   the same date.
```
Comment (serhiy-storchaka) [inline Doc/library/datetime.rst]: Actually, this is not true. See #49766.

**Pull request merged**